### PR TITLE
Updated swagger specs for poweroff on vm/vmss and availability sets list by subscription for 2024/07/01

### PR DESCRIPTION
**Pull request by btyu1**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [X] Other, please clarify:
    - _Update existing version to support new changes_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X]  I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> PowerOff is no longer supported on deallocated vms therefore this pr updates the docs to reflect that. This pr also edits the incorrect Availability Set Expand Parameter list .</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-07-01/virtualMachine.json [hunk 1]
```diff
@@ -1326,7 +1326,7 @@
           "VirtualMachines"
         ],
         "operationId": "VirtualMachines_PowerOff",
-        "description": "The operation to power off (stop) a virtual machine. The virtual machine can be restarted with the same provisioned resources. You are still charged for this virtual machine.",
+        "description": "The operation to power off (stop) a virtual machine. The virtual machine can be restarted with the same provisioned resources. You are still charged for this virtual machine. NOTE: This operation is not allowed on a deallocated virtual machine.",
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-07-01/virtualMachine.json:1329]: This operation is not allowed on a virtual machine that is being deallocated or has already been deallocated.

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-07-01/virtualMachineScaleSet.json [hunk 2]
```diff
@@ -1243,7 +1243,7 @@
           "VirtualMachineScaleSets"
         ],
         "operationId": "VirtualMachineScaleSets_PowerOff",
-        "description": "Power off (stop) one or more virtual machines in a VM scale set. Note that resources are still attached and you are getting charged for the resources. Instead, use deallocate to release resources and avoid charges.",
+        "description": "Power off (stop) one or more virtual machines in a VM scale set. Note that resources are still attached and you are getting charged for the resources. Instead, use deallocate to release resources and avoid charges. Additionally, this operation is not allowed on a VM scale set with a deallocated virtual machine.",
         "parameters": [
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-07-01/virtualMachineScaleSet.json:1247]: Additionally, this operation is not allowed on  virtual machines in a VM scale set that are being deallocated or have already been deallocated

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-07-01/virtualMachineScaleSet.json [hunk 3]
```diff
@@ -3256,7 +3256,7 @@
           "VirtualMachineScaleSetVMs"
         ],
         "operationId": "VirtualMachineScaleSetVMs_PowerOff",
-        "description": "Power off (stop) a virtual machine in a VM scale set. Note that resources are still attached and you are getting charged for the resources. Instead, use deallocate to release resources and avoid charges.",
+        "description": "Power off (stop) a virtual machine in a VM scale set. Note that resources are still attached and you are getting charged for the resources. Instead, use deallocate to release resources and avoid charges. Additionally, this operation is not allowed on a deallocated virtual machine.",
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-07-01/virtualMachineScaleSet.json:3259]: Additionally, this operation is not allowed on a virtual machine that is being deallocated or already has been deallocated.

**Comment by wanlwanl** [comment]
@yuchwang-microsoft  almost OOM for python? can we avoid it? https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4307450&view=logs&j=b7f68bce-9df8-5468-1c9f-74766cfc9a39&t=25fc8ae6-51ed-58e2-b772-f4d054412fe3&l=14032

**Comment by msyyc** [comment]
> @msyyc
> 
> 1. almost OOM for python? can we avoid it? https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4307450&view=logs&j=b7f68bce-9df8-5468-1c9f-74766cfc9a39&t=25fc8ae6-51ed-58e2-b772-f4d054412fe3&l=14032
> 2. there's an error log but mark `"result": "succeeded"` in `generateOutput.json` https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4307450&view=logs&j=b7f68bce-9df8-5468-1c9f-74766cfc9a39&t=25fc8ae6-51ed-58e2-b772-f4d054412fe3&l=23946

1. compute is multiapi package in python which has much bigger size than other packages. We are working on trim it but may take a long time.
2. Will fix it soon.

**Comment by btyu1** [comment]
Work item to address the vmss exposed secrets: [Task 30179919](https://msazure.visualstudio.com/One/_workitems/edit/30179919): Update Swagger specs to remove secret properties from VMSS reponses

**Review by smotwani — APPROVED** [review]

**Review by grizzlytheodore — APPROVED** [review]

**Pull request merged**